File: samples/app/src/main/java/com/ironsource/aura/aircon/sample/config/model/Label.java
package com.ironsource.aura.aircon.sample.config.model;

public class Label {

	private final String mStr;

	private Label(final String str) {
		mStr = str;
	}

	public String getStr() {
		return mStr;
	}

	public static Label from(final String val) {
		return new Label(val);
	}
}

File: samples/app/src/main/java/com/ironsource/aura/airconkt/sample/config/Label.kt
package com.ironsource.aura.airconkt.sample.config

import com.ironsource.aura.airconkt.config.Config
import com.ironsource.aura.airconkt.config.ConfigPropertyFactory
import com.ironsource.aura.airconkt.config.SourceTypeResolver

data class Label(val value: String)

fun labelConfig(block: Config<String, Label>.() -> Unit) =
        ConfigPropertyFactory.from(SourceTypeResolver.string(),
                validator = { it.isNotEmpty() },
                getterAdapter = { Label(it) },
                setterAdapter = { it.value },
                block = block
        )